nncf/openvino/graph/model_transformer.py

```diff
@@ -526,3 +530,42 @@ def _apply_multiply_insertion_transformations(
                 destination_port.replace_source_output(multiply_node.output(0))
 
         return model
+
+    @staticmethod
+    def _apply_update_if_op_subgraph_transformations(
+        model: ov.Model, transformations: List[OVUpdateIfSubgraphCommand]
+    ) -> ov.Model:
+        """
+        Update model subgraph for IF node.
+
+        :param model: Model to update and insert a new subgraph.
+        :param transformations: Transformations with information of If node and an updated subgraph.
+        :return: Original model with an updated subgraph.
+        """
+        name_to_node_mapping = OVModelTransformer._get_name_to_node_mapping(model)
+        for transformation in transformations:
+            subgraph_model = transformation.subgraph_model
+            port_id = transformation.target_point.port_id
+            node_name = transformation.target_point.target_node_name
+            node = name_to_node_mapping[node_name]
+            node.set_function(port_id, subgraph_model)
+        return model
+
+    @staticmethod
+    def _apply_extract_if_subgraph_transformation(
+        model: ov.Model, transformations: List[OVExtractIfSubgraphCommand]
+    ) -> ov.Model:
+        """
+        Extract a model subgraph from If node.
+
+        :param model: Model from which extracts a subgraph.
+        :param transformations: Transformations with information from which
+        If node and input port extract a model subgraph.
+        :return: Model subgraph.
+        """
+        transformation = transformations[-1]
+        name_to_node_mapping = OVModelTransformer._get_name_to_node_mapping(model)
+        ov_node = name_to_node_mapping[transformation.if_node.node_name]
+        if transformation.if_submodel_condition:
+            return ov_node.get_function(0)
+        return ov_node.get_function(1)
```

Comment:
Is it the new `ov.Model` instance or just a link to the object?

Reply:
According to the docs https://docs.openvino.ai/2023.0/classov_1_1op_1_1util_1_1MultiSubGraphOp.html#methods
`return pointer to Model with sub-graph`

Reply:
So that means that we can change the sub-graph `in-place` in the original model.